tools/java/org/unicode/cldr/util/XMLSource.java

```diff
@@ -1747,4 +1764,711 @@ public DtdType getDtdType() {
+        }
+
+        @Override
+        public void startEntity(String name) throws SAXException {
+            Log.logln(LOG_PROGRESS, "startEntity name: " + name);
+        }
+
+        @Override
+        public void endEntity(String name) throws SAXException {
+            Log.logln(LOG_PROGRESS, "endEntity name: " + name);
+        }
+
+        @Override
+        public void startCDATA() throws SAXException {
+            Log.logln(LOG_PROGRESS, "startCDATA");
+        }
+
+        @Override
+        public void endCDATA() throws SAXException {
+            Log.logln(LOG_PROGRESS, "endCDATA");
+        }
+
+        /*
+         * (non-Javadoc)
+         *
+         * @see org.xml.sax.ErrorHandler#error(org.xml.sax.SAXParseException)
+         */
+        @Override
+        public void error(SAXParseException exception) throws SAXException {
+            Log.logln(LOG_PROGRESS || true, "error: " + showSAX(exception));
+            throw exception;
+        }
+
+        /*
+         * (non-Javadoc)
+         *
+         * @see org.xml.sax.ErrorHandler#fatalError(org.xml.sax.SAXParseException)
+         */
+        @Override
+        public void fatalError(SAXParseException exception) throws SAXException {
+            Log.logln(LOG_PROGRESS, "fatalError: " + showSAX(exception));
+            throw exception;
+        }
+
+        /*
+         * (non-Javadoc)
+         *
+         * @see org.xml.sax.ErrorHandler#warning(org.xml.sax.SAXParseException)
+         */
+        @Override
+        public void warning(SAXParseException exception) throws SAXException {
+            Log.logln(LOG_PROGRESS, "warning: " + showSAX(exception));
+            throw exception;
+        }
+    }
+
+    /**
+     * Sets the initial comment, replacing everything that was there.
+     */
+    public XMLSource setInitialComment(String comment) {
+        if (locked) throw new UnsupportedOperationException("Attempt to modify locked object");
+        this.getXpathComments().setInitialComment(comment);
+        return this;
+    }
+
+    public XMLSource addComment(String xpath, String comment, Comments.CommentType type) {
+        if (locked) throw new UnsupportedOperationException("Attempt to modify locked object");
+        Log.logln(LOG_PROGRESS, "ADDING Comment: \t" + type + "\t" + xpath + " \t" + comment);
+        if (xpath == null || xpath.length() == 0) {
+            this.getXpathComments().setFinalComment(
+                CldrUtility.joinWithSeparation(this.getXpathComments().getFinalComment(), XPathParts.NEWLINE,
+                    comment));
+        } else {
+            xpath = CLDRFile.getDistinguishingXPath(xpath, null);
+            this.getXpathComments().addComment(type, xpath, comment);
+        }
+        return this;
+    }
+
+    /**
+     * Get a string value from an xpath.
+     */
+    public String getStringValue(String xpath) {
+            String result = this.getValueAtPath(xpath);
```

Comment:
nitpick: `return this.getValueAtPath(xpath);`

Reply:
Agree, will refractor